### PR TITLE
Fix flag assignment in pal_termios

### LINKED ISSUE #108404: System.IO.Ports.Native termios c_iflag settings ignored?
### Description

Hi, from reading the System.IO.Ports.Native code it seems that the termios c_iflag flags set in line [0] are overwritten by the assignment (!) in line [1], before tcsetattr() is called in [2]. This might lead to termios flags being ignored? Kind regards, Thomas

term.c_iflag &= ~(IMAXBEL|IGNBRK|BRKINT|PARMRK|ISTRIP|INLCR|IGNCR|ICRNL|IXON); // see [0]
...
term.c_iflag = IGNBRK; // see [1]
...
... tcsetattr(fd, TCSANOW, &term) ... // see [2]

[0] https://github.com/dotnet/runtime/blob/main/src/native/libs/System.IO.Ports.Native/pal_termios.c#L446
[1] https://github.com/dotnet/runtime/blob/main/src/native/libs/System.IO.Ports.Native/pal_termios.c#L455
[2] https://github.com/dotnet/runtime/blob/main/src/native/libs/System.IO.Ports.Native/pal_termios.c#L540

### Reproduction Steps

Not sure how to run this.

### Expected behavior

Flags are set, not overwritten.

### Actual behavior

Flags might be overwritten.

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

.NET 8, "Unix".

### Other information

Maybe the assignment should be replaced by setting / clearning a flag instead?

### PR DESCRIPTION
**Pull request by am11**
Fix #108404

### CODE REVIEW

File: src/native/libs/System.IO.Ports.Native/pal_termios.c [hunk 1]
```diff
@@ -437,22 +437,22 @@ int32_t SystemIoPortsNative_TermiosReset(intptr_t handle, int32_t speed, int32_t
...
+    term.c_cflag &= ~(CSIZE | PARENB);
     term.c_cflag |= CS8;
 #endif
-    term.c_cflag |=  (CLOCAL | CREAD);
-    term.c_lflag &= ~((tcflag_t)(ICANON | ECHO | ECHOE | ECHOK | ECHONL | ISIG | IEXTEN ));
+    term.c_cflag |= (CLOCAL | CREAD);
+    term.c_lflag &= ~((tcflag_t)(ICANON | ECHO | ECHOE | ECHOK | ECHONL | ISIG | IEXTEN));
     term.c_oflag &= ~((tcflag_t)(OPOST));
-    term.c_iflag = IGNBRK;
+    term.c_iflag |= IGNBRK;
```
Comment (am11) [inline src/native/libs/System.IO.Ports.Native/pal_termios.c:455]: This is the fix.

Comment (tamberg) [reply src/native/libs/System.IO.Ports.Native/pal_termios.c:455]: Thanks for the quick reply. Note that the same flag is already set in the first line:

term.c_iflag &= ~(IMAXBEL|**IGNBRK**|BRKINT|...

Do you still think, this is the fix?

Comment (am11) [reply src/native/libs/System.IO.Ports.Native/pal_termios.c:455]: I removed it from the first line.

Comment (am11) [reply src/native/libs/System.IO.Ports.Native/pal_termios.c:455]: I can do. Just didn't liked that overloaded/long line at first glance. 😅

Comment (tamberg) [reply src/native/libs/System.IO.Ports.Native/pal_termios.c:455]: Considering https://www.man7.org/linux/man-pages/man3/termios.3.html
```
	cfmakeraw() sets the terminal to something like the "raw" mode ...

	The terminal attributes are set as follows:

           termios_p->c_iflag &= ~(IGNBRK | BRKINT | PARMRK | ISTRIP | INLCR | IGNCR | ICRNL | IXON);
           termios_p->c_oflag &= ~OPOST;
           termios_p->c_lflag &= ~(ECHO | ECHONL | ICANON | ISIG | IEXTEN);
           termios_p->c_cflag &= ~(CSIZE | PARENB);
           termios_p->c_cflag |= CS8;
```

**Comment by janvorli** [comment]
I'll let @wfurt comment on this. I am not sure whether setting the IGNBRK was intentional and just accidentally overwritten the other flags due to the assignment instead of or-ing or whether it should not be set at all as the current change does.

File: src/native/libs/System.IO.Ports.Native/pal_termios.c [hunk 2]
```diff
@@ -437,22 +437,21 @@ int32_t SystemIoPortsNative_TermiosReset(intptr_t handle, int32_t speed, int32_t
...
     {
-        return  -1;
+        return -1;
     }
 
 #if HAVE_CFMAKERAW
     cfmakeraw(&term);
 #else
-    term.c_iflag &= ~(IMAXBEL|IGNBRK|BRKINT|PARMRK|ISTRIP|INLCR|IGNCR|ICRNL|IXON);
+    term.c_iflag &= ~(IMAXBEL | IGNBRK | BRKINT | PARMRK | ISTRIP | INLCR | IGNCR | ICRNL | IXON);
```
Comment (am11) [inline src/native/libs/System.IO.Ports.Native/pal_termios.c:446]: @janvorli, it is set here. The redundant line, which was overwriting the flag was deleted.

Comment (am11) [inline src/native/libs/System.IO.Ports.Native/pal_termios.c:446]: @janvorli, it is set here. The redundant line 455, which was overwriting all the flags, was deleted.

**Review by wfurt — APPROVED** [review]
LGTM

**Comment by janvorli** [comment]
The runtime-dev-innerloop seems to be failing on many if not all PRs.

**Pull request merged**